application/backend/app/services/dataset_service.py

```diff
@@ -373,3 +374,31 @@ def _get_image_path(item: DatasetItem) -> str:
             get_dataset_items=_get_dataset_items,
             get_image_path=_get_image_path,
         )
+
+    def save_revision(self, project_id: UUID, dataset: dm.Dataset) -> None:
+        """
+        Saves the dataset as a new revision.
+
+        Creates a new dataset revision entry in the database and exports the dataset
+        to a zip file in the project's revisions directory.
+
+        Args:
+            project_id: The UUID of the project to save the revision for.
+            dataset: The Datumaro dataset to export.
+
+        Returns:
+            None
+        """
+        revision_repo = DatasetRevisionRepository(db=self.db_session)
+        revision_db = revision_repo.save(
+            DatasetRevisionDB(
+                project_id=str(project_id),
+            )
+        )
+        revision_path = self.projects_dir / str(project_id) / "revisions" / revision_db.id
```

Comment:
`"revisions"` -> `"dataset_revisions"`, to avoid ambiguity with model revisions.